netkan/netkan/common.py

```diff
@@ -52,3 +70,89 @@ def download_stream_to_file(download_url: str, dest_file: IO[bytes]) -> None:
                       timeout=60) as req:
         for chunk in req.iter_content(chunk_size=8192):
             dest_file.write(chunk)
+
+
+class BaseMessageHandler:
+    STRATEGY_OPTION = 'ours'
+    game: Game
+
+    def __init__(self, game: Game) -> None:
+        self.game = game
+
+    # Apparently gitpython can be leaky on long running processes
+    # we can ensure we call close on it and run our handler inside
+    # a context manager
+    def __enter__(self) -> 'BaseMessageHandler':
+        if not self.repo.is_active_branch('master'):
+            self.repo.checkout_branch('master')
+        self.repo.pull_remote_branch(
+            'master', strategy_option=self.STRATEGY_OPTION)
+        return self
+
+    def __exit__(self, exc_type: Type[BaseException],
+                 exc_value: BaseException, traceback: TracebackType) -> None:
+        self.repo.close_repo()
+
+    @property
+    def repo(self) -> Union[CkanMetaRepo, NetkanRepo]:
+        raise NotImplementedError
+
+    def append(self, message: Message) -> None:
+        raise NotImplementedError
+
+    def process_messages(self) -> None:
+        raise NotImplementedError
+
+    def sqs_delete_entries(self) -> List[DeleteMessageBatchRequestEntryTypeDef]:
+        raise NotImplementedError
+
+
+class QueueHandler:
+    common: SharedArgs
+    _game_handlers: Dict[str, BaseMessageHandler]
+    _handler_class: Type[BaseMessageHandler] = BaseMessageHandler
+
+    def __init__(self, common: SharedArgs) -> None:
+        self.common = common
```

Comment:
Open to refactoring it in a future PR. A lot has changed since the initial codebase was formed. We definitely can't use `args` though :slightly_smiling_face:

Reply:
But this is all new code. The `Game` class is completely new. Getting the name right now makes more sense to me than creating a poor name and deferring a good name to the indefinite future.

Reply:
The future won't be indefinite, and I agree with you. I just would rather get this changeset out and in operation first. As it'll be a much simpler change to test once it's scoped to that.